### PR TITLE
Add a netcore50aot configuration for System.Resources.ResourceManager

### PR DESCRIPTION
**Pull request by mellinoe**
Includes some new code which I have cleaned up and formatted in the second commit.

@weshaggard

### CODE REVIEW

File: src/System.Resources.ResourceManager/src/System/Resources/MissingManifestResourceException.cs [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+{
+    public class MissingManifestResourceException : Exception
+    {
+        public MissingManifestResourceException()
+            : base(SR.Arg_MissingManifestResourceException)
+        {
+            HResult = HResults.COR_E_MISSINGMANIFESTRESOURCE;
+        }
+
+        public MissingManifestResourceException(String message)
```
Comment (justinvp) [inline src/System.Resources.ResourceManager/src/System/Resources/MissingManifestResourceException.cs]: Nit: `String` -> `string`

File: src/System.Resources.ResourceManager/src/System/Resources/MissingManifestResourceException.cs [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+            : base(message)
+        {
+            HResult = HResults.COR_E_MISSINGMANIFESTRESOURCE;
+        }
+
+        public MissingManifestResourceException(String message, Exception inner)
```
Comment (justinvp) [inline src/System.Resources.ResourceManager/src/System/Resources/MissingManifestResourceException.cs]: Nit: `String` -> `string`

File: src/System.Resources.ResourceManager/src/System/Resources/NeutralResourcesLanguageAttribute.cs [hunk 3]
```diff
@@ -0,0 +1,26 @@
...
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+namespace System.Resources
+{
+    [AttributeUsage(AttributeTargets.Assembly, AllowMultiple = false)]
+    public sealed class NeutralResourcesLanguageAttribute : Attribute
+    {
+        private readonly string _culture;
+
+        public NeutralResourcesLanguageAttribute(String cultureName)
```
Comment (justinvp) [inline src/System.Resources.ResourceManager/src/System/Resources/NeutralResourcesLanguageAttribute.cs]: Nit: `String` -> `string`

File: src/System.Resources.ResourceManager/src/System/Resources/SatelliteContractVersionAttribute.cs [hunk 4]
```diff
@@ -0,0 +1,26 @@
...
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+namespace System.Resources
+{
+    [AttributeUsage(AttributeTargets.Assembly, AllowMultiple = false)]
+    public sealed class SatelliteContractVersionAttribute : Attribute
+    {
+        private readonly string _version;
+
+        public SatelliteContractVersionAttribute(String version)
```
Comment (justinvp) [inline src/System.Resources.ResourceManager/src/System/Resources/SatelliteContractVersionAttribute.cs]: Nit: `String` -> `string`

**Comment by mellinoe** [comment]
Oops, thought I did a find/replace on the whole project, but apparently I missed a bunch of `String` occurrences. Fixing those.

**Comment by stephentoub** [comment]
Test Innerloop Windows_NT Release Build and Test please
Test Innerloop Windows_NT Debug Build and Test please
Test Innerloop CentOS7.1 Release Build and Test please
Test Innerloop CentOS7.1 Debug Build and Test please

File: src/System.Resources.ResourceManager/src/netcore50aot/project.json [hunk 5]
```diff
@@ -0,0 +1,15 @@
...
+  "frameworks": {
+    "netcore50": {
+      "dependencies": {
+        "System.Runtime": "4.0.20",
+        "System.Diagnostics.Debug": "4.0.0",
+        "System.Diagnostics.Tools": "4.0.0",
+        "System.Diagnostics.Tracing": "4.0.0",
+        "System.Globalization": "4.0.0",
+        "System.Reflection": "4.0.10",
+        "Microsoft.TargetingPack.Private.NETNative": "1.0.0-rc2-23707"
```
Comment (weshaggard) [inline src/System.Resources.ResourceManager/src/netcore50aot/project.json]: Please sort these dependencies.

**Comment by weshaggard** [comment]
Looks like the failure is related to your changes:
http://dotnet-ci.cloudapp.net/job/dotnet_corefx/job/windows_nt_debug_prtest/868/console
```
System\Resources\ResourceManager.cs(73,70): error CS0103: The name 'SR' does not exist in the current context [d:\j\workspace\dotnet_corefx\windows_nt_debug_prtest\src\System.Resources.ResourceManager\src\System.Resources.ResourceManager.csproj]
13:33:21 System\Resources\ResourceManager.cs(73,60): error CS0103: The name 'SR' does not exist in the current context [d:\j\workspace\dotnet_corefx\windows_nt_debug_prtest\src\System.Resources.ResourceManager\src\System.Resources.ResourceManager.csproj]
13:33:21 System\Resources\MissingManifestResourceException.cs(9,20): error CS0103: The name 'SR' does not exist in the current context 
```

**Comment by weshaggard** [comment]
LGTM once the CI is green

**Comment by mellinoe** [comment]
I had messed something up in my second commit. Tried to exclude the string resources from the configurations that don't need them, but messed up the .NET Native configuration in the process. Should be fixed now.

**Comment by KrzysztofCwalina** [comment]
Could you add the same to System.Resources.Reader and System.Resources.Writer?

**Comment by mellinoe** [comment]
> Could you add the same to System.Resources.Reader and System.Resources.Writer?

I think we can use the same configuration for S.R.Reader and Writer, so I don't think we need to do anything special for them. They don't have any platform dependencies.

**Comment by KrzysztofCwalina** [comment]
S.R.R and S.R.W are just refactored S.R.RW, so yes we can use the same configuration. But, I will be removing/deleting S.R.RW later this week, so this PR needs to be ported to S.R.R and S.R.W. @mellinoe, would you be willing to do this port?

**Comment by mellinoe** [comment]
This PR is for ResourceManager, we aren't planning any refactoring for this library, right?

**Comment by KrzysztofCwalina** [comment]
Ah, sorry. No, we are not removing/refactoring ResourceManager.

**Comment by mellinoe** [comment]
Innerloop CentOS7.1 Release Build and Test
Innerloop Ubuntu Release Build and Test

**Comment by mellinoe** [comment]
Innerloop CentOS7.1 Release Build and Test

**Comment by mellinoe** [comment]
The CI doesn't seem to want to restart CentOS release, but I'm going to merge anyways. This change won't affect any of the test runs anyways, and the other CentOS build succeeded.

**Pull request merged**